PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = ['account.move']
-    _inherit = ['account.move.line']
-    _inherit = ["product.template"]
-    _inherit = ['purchase.order.line']
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ["stock.move"]
-    _inherit = ['stock.valuation.layer']

Total Changes: 19

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock_landed_costs/models/account_move.py
+++ b/addons/stock_landed_costs/models/account_move.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     landed_costs_ids = fields.One2many('stock.landed.cost', 'vendor_bill_id', string='Landed Costs')
     landed_costs_visible = fields.Boolean(compute='_compute_landed_costs_visible')
@@ -53,7 +53,7 @@ class AccountMove(models.Model):
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ['account.move.line']
+    _inherit = 'account.move.line'
 
     product_type = fields.Selection(related='product_id.type', readonly=True)
     is_landed_costs_line = fields.Boolean()

--- a/addons/stock_landed_costs/models/product.py
+++ b/addons/stock_landed_costs/models/product.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class ProductTemplate(models.Model):
-    _inherit = ["product.template"]
+    _inherit = "product.template"
 
     landed_cost_ok = fields.Boolean('Is a Landed Cost', help='Indicates whether the product is a landed cost: when receiving a vendor bill, you can allocate this cost on preceding receipts.')
     split_method_landed_cost = fields.Selection(SPLIT_METHOD, string="Default Split Method",

--- a/addons/stock_landed_costs/models/purchase.py
+++ b/addons/stock_landed_costs/models/purchase.py
@@ -2,7 +2,7 @@ from odoo import api,models
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = ['purchase.order.line']
+    _inherit = 'purchase.order.line'
 
     def _prepare_account_move_line(self, move=False):
         res = super()._prepare_account_move_line(move)

--- a/addons/stock_landed_costs/models/res_company.py
+++ b/addons/stock_landed_costs/models/res_company.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     lc_journal_id = fields.Many2one('account.journal')

--- a/addons/stock_landed_costs/models/res_config_settings.py
+++ b/addons/stock_landed_costs/models/res_config_settings.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     lc_journal_id = fields.Many2one('account.journal', string='Default Journal', related='company_id.lc_journal_id', readonly=False)

--- a/addons/stock_landed_costs/models/stock_landed_cost.py
+++ b/addons/stock_landed_costs/models/stock_landed_cost.py
@@ -18,6 +18,7 @@ SPLIT_METHOD = [
 
 
 class StockLandedCost(models.Model):
+    _name = 'stock.landed.cost'
     _description = 'Stock Landed Cost'
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _order = 'date desc, id desc'
@@ -339,6 +340,7 @@ class StockLandedCost(models.Model):
 
 
 class StockLandedCostLines(models.Model):
+    _name = 'stock.landed.cost.lines'
     _description = 'Stock Landed Cost Line'
 
     name = fields.Char('Description')
@@ -369,6 +371,7 @@ class StockLandedCostLines(models.Model):
 
 
 class StockValuationAdjustmentLines(models.Model):
+    _name = 'stock.valuation.adjustment.lines'
     _description = 'Valuation Adjustment Lines'
 
     name = fields.Char(

--- a/addons/stock_landed_costs/models/stock_move.py
+++ b/addons/stock_landed_costs/models/stock_move.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class StockMove(models.Model):
-    _inherit = ["stock.move"]
+    _inherit = "stock.move"
 
     def _get_stock_valuation_layer_ids(self):
         self.ensure_one()

--- a/addons/stock_landed_costs/models/stock_valuation_layer.py
+++ b/addons/stock_landed_costs/models/stock_valuation_layer.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class StockValuationLayer(models.Model):
-    _inherit = ['stock.valuation.layer']
+    _inherit = 'stock.valuation.layer'
 
     stock_landed_cost_id = fields.Many2one('stock.landed.cost', 'Landed Cost')
